src/ConcordiumNetSdk/Transactions/SignedAccountTransaction.cs

```diff
@@ -0,0 +1,143 @@
+        )
+        {
+            this._header = header;
+            this._payload = payload;
+            this._signature = signature;
+        }
+
+        /// <summary>
+        /// Creates a new instance of a signed transaction.
+        /// </summary>
+        /// <param name="sender">Address of the sender of the transaction.</param>
+        /// <param name="nonce">Account nonce to use for the transaction.</param>
+        /// <param name="expiry">Expiration time of the transaction.</param>
+        /// <param name="payload">Payload to send to the node.</param>
+        /// <param name="signer">The signer to use for signing the transaction.</param>
+        public static SignedAccountTransaction<T> Create(
+            AccountAddress sender,
+            AccountNonce nonce,
+            Expiry expiry,
+            AccountTransactionPayload<T> payload,
+            ITransactionSigner transactionSigner
+        )
+        {
+            // Get the serialized payload.
+            byte[] serializedPayload = payload.GetBytes();
+            UInt32 serializedPayloadSize = (UInt32)serializedPayload.Length;
+
+            // Compute the energy cost.
+            UInt64 txSpecificCost = payload.GetBaseEnergyCost();
+            UInt64 energyCost = CalculateEnergyCost(
+                transactionSigner.GetSignatureCount(),
+                txSpecificCost,
+                AccountTransactionHeader.BytesLength,
+                serializedPayloadSize
+            );
+
+            // Create the header.
+            var header = new AccountTransactionHeader(
+                sender,
+                nonce,
+                expiry,
+                energyCost,
+                serializedPayloadSize
+            );
+
+            // Construct the serialized payload and its digest for signing.
+            byte[] serializedHeaderAndTxPayload = header
+                .GetBytes()
+                .Concat(serializedPayload)
+                .ToArray();
+            byte[] signDigest = SHA256.Create().ComputeHash(serializedHeaderAndTxPayload);
+
+            // Sign it.
+            AccountTransactionSignature signature = AccountTransactionSignature.Create(
+                transactionSigner,
+                signDigest
+            );
+
+            return new SignedAccountTransaction<T>(header, payload, signature);
+        }
+
+        /// <summary>
+        /// Calculates the energy cost associated with processing the transaction.
+        /// </summary>
+        /// <param name="signatureCount">The number of signatures.</param>
+        /// <param name="txSpecificCost">The transaction specific cost.</param>
+        /// <param name="headerSize">The size of the header in bytes.</param>
+        /// <param name="payloadSize">The size of the payload in bytes.</param>
+        private static ulong CalculateEnergyCost(
+            UInt32 signatureCount,
+            UInt64 txSpecificCost,
+            UInt32 headerSize,
```

Comment:
Why is this passed as a parameter? Isn't there a constant defined for this already?

A function to compute the cost of a transaction (in NRG) should be public. Maybe not this exact one, but some way to compute the cost before sending or signing the transaction.

Reply:
This is a private static method and the reason for keeping it like that was to make it explicit that the energy cost is computed based on the header size. There is no drawback from doing so IMO, but let me know if I should change it.

Where do you reckon such a function should be? Do we want an actual function or should I perhaps expose the cost as a readonly field in the header?